engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -1976,6 +1977,12 @@ private void advanceStop(final VMInstanceVO vm, final boolean cleanUpEvenIfUnabl
                     s_logger.warn("Unable to actually stop " + vm + " but continue with release because it's a force stop");
                     vmGuru.finalizeStop(profile, answer);
                 }
+            } else {
+                if (VirtualMachine.Type.ConsoleProxy == vm.getType() || VirtualMachine.Type.SecondaryStorageVm == vm.getType()) {
```

Comment:
I think I expressed myself badly. I meant you to create the collection as a constant and then use it, like:

```java
private final Set<VirtualMachine.Type> <insertNameHere> = new HashSet<>(Arrays.asList(VirtualMachine.Type.ConsoleProxy, VirtualMachine.Type.SecondaryStorageVm));
...
if (<insertNameHere>.contains(vm.getType())) {
```